src/Resources/contao/library/Contao/InsertTags.php

```diff
@@ -120,7 +120,10 @@ protected function doReplace($strBuffer, $blnCache)
 				{
 					/** @var FragmentHandler $fragmentHandler */
 					$fragmentHandler = \System::getContainer()->get('fragment.handler');
-					$strBuffer .= $fragmentHandler->render(new ControllerReference('contao.controller.insert_tags:renderAction', ['insertTag' => '{{' . $strTag . '}}']), 'esi');
+					$strBuffer .= $fragmentHandler->render(new ControllerReference('contao.controller.insert_tags:renderAction',
+						['insertTag' => '{{' . $strTag . '}}'],
+						['pageId' => $objPage->id, 'request' => \Environment::get('request')]
```

Comment:
I think `request` is better. `Referrer` is an HTTP header which does not apply here. It's a request query attribute. It doesn't really matter, I just found `request` to be the better name which is why I chose it.